Makefile

```diff
@@ -102,12 +102,26 @@ push-chart:
 	helm push _output/charts/karmada-chart-${VERSION}.tgz oci://docker.io/karmada
 	helm push _output/charts/karmada-operator-chart-${VERSION}.tgz oci://docker.io/karmada
 
+COLOR_GOTEST_REGISTRY:=github.com/rakyll/gotest
+COLOR_GOTEST_VERSION:=aeb9f1f4739020c60963f21eec2e65672307a9ac
+COLOR_GOTEST_PALETTE:=hired,higreen
+COLOR_GOTEST_ENABLED?=
+GOTEST=go test
+
+.PHONY: install_gotest
+install_gotest:
+ifdef COLOR_GOTEST_ENABLED 
+	go install ${COLOR_GOTEST_REGISTRY}@${COLOR_GOTEST_VERSION}
+GOTEST=gotest
+endif
+
 .PHONY: test
-test:
+test:export GOTEST_PALETTE=${COLOR_GOTEST_PALETTE}
+test: install_gotest
```

Comment:
I'm not familiar with this usage, why use two targets here?

Reply:
The first one is to export the environment only in `test`.  

It seems not necessary to do this. I have removed it, and now it could be set straightly as an environment.  

Now the performance would be like:  
- If `COLOR_GOTEST_ENABLED` environment is not set, `make test` would use `go test` straightly.  
- If `COLOR_GOTEST_ENABLED` is set but `GOTEST_PALETTE` is not set, `make test` would color the output using `hired` for `FAIL` and `higreen` for `PASS` in default.  
- If both `COLOR_GOTEST_ENABLED` and `GOTEST_PALETTE` are set, `make test` would color the output by the declared color by `GOTEST_PALETTE` value, the value is `<color for FAIL>,<color for PASS>`

Reply:
I can't wait to enable it in CI, I guess what we need to do is:
```diff
diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index 45a4982fb..2c321e4fb 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -88,7 +88,9 @@ jobs:
         with:
           go-version: 1.20.4
       - name: make test
-        run: make test
+        run: |
+          export COLOR_GOTEST_ENABLED=true
+          make test
       - name: Upload coverage to Codecov
         # Prevent running from the forked repository that doesn't need to upload coverage.
         # In addition, running on the forked repository would fail as missing the necessary secret.
```

Can you help to add this in this PR? so that other guys can see the magic :)

Reply:
Done.  

```diff
diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index 45a4982f..4daf2c55 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -80,6 +80,8 @@ jobs:
     name: unit test
     needs: build
     runs-on: ubuntu-22.04
+    env:
+      COLOR_GOTEST_ENABLED: enabled
     steps:
       - name: checkout code
         uses: actions/checkout@v3
```

We just need to add an environment for `unit test` job. Hope you enjoy the colorful UT! :)  
https://github.com/karmada-io/karmada/actions/runs/5700866002/job/15451158398